lucene/core/src/java/org/apache/lucene/index/MergePolicy.java

```diff
@@ -526,6 +526,19 @@ public abstract MergeSpecification findForcedMerges(
   public abstract MergeSpecification findForcedDeletesMerges(
       SegmentInfos segmentInfos, MergeContext mergeContext) throws IOException;
 
+  /**
+   * Identifies merges that we want to execute (synchronously) on commit. By default, do not synchronously merge on commit.
+   *
+   * If a returned {@link OneMerge} includes a segment already included in a registered merge, then the commit will fail.
```

Comment:
Can you say what exception will be thrown in that case (or add an `@throws`, below)?